### ISSUE TITLE
[UI] Migrate React Select Wrapper to Functional Component with MUI v5 Styling and Sistent Component use

### ISSUE THREAD

**Issue by SAHU-01** [meshery/meshery#12120]
### Current Behavior
This [ReactSelectWrapper](https://github.com/meshery/meshery/blob/268c1667a6ab6a8bbdcbd960df20023d8ecbfb3a/ui/components/ReactSelectWrapper.js#L147) Component is a class-based component and it uses MUI v4's styling. These need to be updated for improved performance and consistency.

### Desired Behavior
- [ ]  migrate class-based component to a functional component
- [ ]  use [Sistent](https://github.com/layer5io/sistent) components for  MUI v4 to MUI v5 migration

Sistent is our design system that uses MUI v5, you can also refer to this [doc](https://docs.meshery.io/project/contributing/contributing-ui-sistent) to understand how to use Sistent.

### Acceptance Tests
The component must be updated without introducing any breaking changes in functionality or theming across all instances where it is used.
---

### Contributor [Guides](https://docs.meshery.io/project/contributing) and [Handbook](https://layer5.io/community/handbook)

- 🎨 Wireframes and [designs for Meshery UI](https://www.figma.com/file/SMP3zxOjZztdOLtgN4dS2W/Meshery-UI) in Figma [(open invite)](https://www.figma.com/team_invite/redeem/qJy1c95qirjgWQODApilR9)
- 🖥 [Contributing to Meshery UI](https://docs.meshery.io/project/contributing/contributing-ui)
- 🙋🏾🙋🏼 Questions: [Discussion Forum](http://discuss.meshery.io) and [Community Slack](https://slack.meshery.io)

**Comment by Mandeep56Singh** [comment]
I want to work on this

**Comment by THEBOSS0369** [comment]
Hey @SAHU-01 ! Can i work on this?! Thanks

**Comment by SAHU-01** [comment]
@Mandeep56Singh and @THEBOSS0369  could you confirm if you were successful in setting up the site locally, accordingly, I'll be able to break down the task and assign it to both of you? If not, please let me know as well.

**Comment by THEBOSS0369** [comment]
Hey @SAHU-01 ! No i haven't been able to setup the serve locally yet please see this ss. 
I have checked that have required lib like jekyll ruby gem go node. Neither i found any soltuion for this on discussion forums.
Thanks

![image](https://github.com/user-attachments/assets/09433472-7b3c-4b68-b8f4-fbcf15796937)
![image](https://github.com/user-attachments/assets/554fbb7a-b380-41b6-a484-f692617eab73)

**Comment by SAHU-01** [comment]
> Hey @SAHU-01 ! No i haven't been able to setup the serve locally yet please see this ss. I have checked that have required lib like jekyll ruby gem go node. Neither i found any soltuion for this on discussion forums. Thanks
> 
> ![image](https://private-user-images.githubusercontent.com/145831712/371822601-09433472-7b3c-4b68-b8f4-fbcf15796937.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc1ODgzMTgsIm5iZiI6MTcyNzU4ODAxOCwicGF0aCI6Ii8xNDU4MzE3MTIvMzcxODIyNjAxLTA5NDMzNDcyLTdiM2MtNGI2OC1iOGY0LWZiY2YxNTc5NjkzNy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyOVQwNTMzMzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1mNTk0YWZhOGQxYzkwYmUzZTVlN2UwNGE0YWRlMTNhNGIzMzgwYTljOWJhYjE5MjIyMGRiMTU5ZTk5OTEzZmFmJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.TMsgSHQCLjne-ljuBc8kbnDSdi8RJsfxtvV4EXr0zvQ) ![image](https://private-user-images.githubusercontent.com/145831712/371822647-554fbb7a-b380-41b6-a484-f692617eab73.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc1ODgzMTgsIm5iZiI6MTcyNzU4ODAxOCwicGF0aCI6Ii8xNDU4MzE3MTIvMzcxODIyNjQ3LTU1NGZiYjdhLWIzODAtNDFiNi1hNDg0LWY2OTI2MTdlYWI3My5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwOTI5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDkyOVQwNTMzMzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1lNjg0NTU0NThlOTBmOGM5ZTQxMGU4ZThiYmE1N2QxM2FlNWY1YTI0ZTM5OTBhN2EwODhiMjIwYzRkOGZkNzVkJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.JVhbnLJKokDt_RS1wofO4n2zdZ7FxcIG8kFN3yGnJCA)
> 
> Note: **Meshery.io** works without any issue on my local system.

@THEBOSS0369 to work on this issue you need to setup and run `meshery/meshery` repo locally, to do so you can follow [this](https://docs.meshery.io/project/contributing/contributing-ui) documentation and let me know over this thread or on slack if you get stuck.

The screenshot you shared was for `meshery/docs` not `meshery/ui`. [This](https://docs.meshery.io/project/contributing/contributing-docs) documenttion might help you with `meshery/docs`, lest you can share the issue in the documentation channel and I'm sure we can help you resolve it.

Once you are successful in the setup, I can assign you this issue.

**Comment by SankarshanMishra** [comment]
I want to solve this issue could you please assign this to me.

**Comment by SAHU-01** [comment]
@SankarshanMishra could you confirm if you've been successful in setting up `meshery` locally? Once, you do, I'll assign you with this issue.

**Comment by Mandeep56Singh** [comment]
@SAHU-01  yesh, I already have meshery running in my local system.

**Comment by SAHU-01** [comment]
That's awesome, @Mandeep56Singh! I'll be assigning you this issue since you were the first to confirm your availability. I noticed you've also commented on a few other issues. To keep things fair for everyone, once you complete this issue and your PR is merged, feel free to ask for another assignment.

> @SAHU-01 yesh, I already have meshery running in my local system.

**Comment by SankarshanMishra** [comment]
> @SankarshanMishra could you confirm if you've been successful in setting up `meshery` locally? Once, you do, I'll assign you with this issue.

yes, I also have it installed, okay since you have assigned it to someone else I will look for different similar 
issues.

**Comment by THEBOSS0369** [comment]
Hey @SAHU-01 ! Same goes for me! I will look for another Issue. Thanks

**Comment by Mandeep56Singh** [comment]
@SAHU-01  I have converted class component to functional, and now migrating mui v4 to mui v5, but there's a problem.
mui v5 major change was style engine, so I thought of using styled api instead of with styles but with this I am not able to use custom styles for colors, and some styling is also changed. 
should I hardcode colors ? 
I have seen hardcoded colors has used in login.js where styled api is used.
I would appreciate your assistance.

**Comment by Mandeep56Singh** [comment]
@SAHU-01 I have completed all task, Just one thing remain, its about color
previously in v4, color:textSecondary was being used, I am not sure which one would be equivalent to this color.
In v5, useTheme hook is used for theme color variables, Please tell where to locate this color and other colors.

**Comment by SAHU-01** [comment]
@Mandeep56Singh you need to use Sistent tokens here rather than hexcodes or harded colors.

**Comment by Mandeep56Singh** [comment]
@SAHU-01 kindly review the PR
